api/python/t4/api.py

```diff
@@ -266,26 +266,29 @@ def create_str(self, pkg_info):
                         f"{self._fmt_str(size_str, 15).rstrip(' ')}\t\n")
             return out
 
+    # get_package_registry may include '/.quilt', which Package.browse does not expect
     if registry is None or registry == 'local':
-        base_registry = get_package_registry(None)
+        package_registry = None
     else:
-        base_registry = get_package_registry(fix_url(registry))
+        package_registry = fix_url(registry)
 
-    named_packages = base_registry + '/named_packages'
+    base_registry = get_package_registry(package_registry)
+    registry = base_registry.rstrip('.quilt/')
+    named_packages_path = urlparse(base_registry + '/named_packages')
+    registry_scheme = named_packages_path.scheme
 
     pkg_info = []
 
-    registry_url = urlparse(named_packages)
-    if registry_url.scheme == 'file':
-        registry_dir = pathlib.Path(parse_file_url(registry_url))
+    if registry_scheme == 'file':
+        registry_dir = pathlib.Path(parse_file_url(named_packages_path))
```

Comment:
this should be named `named_packages_dir` not `registry_dir`

Reply:
Done.